pkg/kubelet/cm/dra/manager.go

```diff
@@ -134,27 +137,42 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 			if pluginName == "" {
 				pluginName = resourceClaim.Status.DriverName
 			}
+			batches[pluginName] = append(batches[pluginName],
+				&drapb.Claim{
+					Namespace:      resourceClaim.Namespace,
+					Uid:            string(resourceClaim.UID),
+					Name:           resourceClaim.Name,
+					ResourceHandle: resourceHandle.Data,
+				})
+		}
+		claimInfos[resourceClaim.UID] = claimInfo
+	}
 
-			// Call NodePrepareResource RPC for each resourceHandle
-			client, err := dra.NewDRAPluginClient(pluginName)
-			if err != nil {
-				return fmt.Errorf("failed to get DRA Plugin client for plugin name %s, err=%+v", pluginName, err)
+	for pluginName, claims := range batches {
+		// Call NodePrepareResources RPC for all resource handles.
+		client, err := dra.NewDRAPluginClient(pluginName)
+		if err != nil {
+			return fmt.Errorf("failed to get DRA Plugin client for plugin name %s: %v", pluginName, err)
+		}
+		response, err := client.NodePrepareResources(context.Background(), &drapb.NodePrepareResourcesRequest{Claims: claims})
+		if err != nil {
+			// General error unrelated to any particular claim.
+			return fmt.Errorf("NodePrepareResources failed: %v", err)
+		}
+		for claimUID, result := range response.Claims {
+			reqClaim := lookupClaimRequest(claims, claimUID)
+			if reqClaim == nil {
+				return fmt.Errorf("NodePrepareResources returned result for unknown claim UID %s", claimUID)
 			}
-			response, err := client.NodePrepareResource(
-				context.Background(),
-				resourceClaim.Namespace,
-				resourceClaim.UID,
-				resourceClaim.Name,
-				resourceHandle.Data)
-			if err != nil {
-				return fmt.Errorf("NodePrepareResource failed, claim UID: %s, claim name: %s, resource handle: %s, err: %+v",
-					resourceClaim.UID, resourceClaim.Name, resourceHandle.Data, err)
+			if result.Error != "" {
+				return fmt.Errorf("NodePrepareResources failed for claim %s/%s: %s", reqClaim.Namespace, reqClaim.Name, err)
```

Comment:
There' something still strange to me about this `Error` field. It is solely for reporting here? Or is it used / interpreted elsewhere?

Reply:
It's only checked here. The content of the `Error` field  then only gets passed on as part of the error created here... except that `err` should have been `result.Error`! Fixed.